### PR TITLE
Replace select images with SVG diagrams or clearer images

### PR DESCRIPTION
**Pull request by yellowjaguar5**
- minor text updates
- revamp sensor pages with cleaner images
- add wiring diagrams

This is the bulk of the changes split from #200.

### CODE REVIEW

File: source/docs/hardware/sensors/proximity-switches.rst [hunk 1]
```diff
@@ -17,49 +17,59 @@ Types of proximity switches
...
 
-Photoelectric proximity switches
+Photoelectric Proximity Switches
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. todo:: add picture
+.. image:: images/proximity-switches-hardware/ir-digital-obstacle-sensor-to-roborio.svg
 
-Photoelectric proximity switches are the last type of no-contact proximity switch in widespread use in FRC.  Photoelectric proximity switches contain a light source (usually an IR laser) and a photoelectric sensor that activates the switch when the detected light (which bounces off of the sensor target) exceeds a given threshold.
+.. image:: images/proximity-switches-hardware/ir-beam-break-sensor-to-roborio.svg
```
Comment (Daltz333) [inline source/docs/hardware/sensors/proximity-switches.rst:62]: Move this below the paragraph for better image flow

File: source/docs/hardware/sensors/proximity-switches.rst [hunk 2]
```diff
@@ -17,49 +17,59 @@ Types of proximity switches
...
+
+Photoelectric proximity switches are another type of no-contact proximity switch in widespread use in FRC.  Photoelectric proximity switches contain a light source (usually an IR laser) and a photoelectric sensor that activates the switch when the detected light (which bounces off of the sensor target) exceeds a given threshold. One such sensor is the `IR Obstacle Avoidance Module <https://www.electrodragon.com/product/infraredir-obstacle-avoidance-sensor-moduleadjust-distance/>`__ pictured above.
```
Comment (Daltz333) [inline source/docs/hardware/sensors/proximity-switches.rst:64]:
```suggestion
Photoelectric proximity switches are another type of no-contact proximity switch in widespread use in FRC.  Photoelectric proximity switches contain a light source (usually an IR laser) and a photoelectric sensor that activates the switch when the detected light (which bounces off of the sensor target) exceeds a given threshold. One such sensor is the `IR Obstacle Avoidance Module <https://www.electrodragon.com/product/infraredir-obstacle-avoidance-sensor-moduleadjust-distance/>`__ pictured below.
```

File: source/docs/hardware/sensors/encoders-hardware.rst [hunk 3]
```diff
@@ -27,25 +27,32 @@ These encoders vary in how they are mounted to the mechanism in question.  In ad
...
 
-Shafted encoders have a sealed body with a shaft protruding out of it that must be coupled rotationally to a mechanism.  This is often done with a helical beam coupling, or, more cheaply, with a length of flexible tubing (such as surgical tubing or pneumatic tubing), fastened with cable ties and/or adhesive at either end.  Many commercial off-the-shelf FRC gearboxes have purpose-built mounting points for shafted encoders, such as the popular `Grayhill 63r <http://www.grayhill.com/assets/1/7/Opt_Encoder_63R.pdf>`__.
+Shafted encoders have a sealed body with a shaft protruding out of it that must be coupled rotationally to a mechanism.  This is often done with a helical beam coupling, or, more cheaply, with a length of flexible tubing (such as surgical tubing or pneumatic tubing), fastened with cable ties and/or adhesive at either end.  Many commercial off-the-shelf FRC gearboxes have purpose-built mounting points for shafted encoders, such as the popular `Grayhill 63r <http://www.grayhill.com/assets/1/7/Opt_Encoder_63R.pdf>`__, pictured above.
 
 On-shaft encoders
 ^^^^^^^^^^^^^^^^^
 
-.. todo:: add picture
+.. image:: images/encoders-hardware/cui-amt102.png
```
Comment (Daltz333) [inline source/docs/hardware/sensors/encoders-hardware.rst:37]: Is it possible to have a SVG picture here, it's a bit jarring. If not, it's fine.

File: source/docs/hardware/sensors/encoders-hardware.rst [hunk 4]
```diff
@@ -27,25 +27,32 @@ These encoders vary in how they are mounted to the mechanism in question.  In ad
 Shafted encoders
 ^^^^^^^^^^^^^^^^
 
-.. todo:: add picture
+.. image:: images/encoders-hardware/grayhill-63r.jpg
```
Comment (Daltz333) [inline source/docs/hardware/sensors/encoders-hardware.rst:30]: Is it possible to have a SVG picture here, it's a bit jarring. If not, it's fine.

Comment (yellowjaguar5) [reply source/docs/hardware/sensors/encoders-hardware.rst]: Added in 8330140.

File: source/docs/hardware/sensors/encoders-hardware.rst (hunk 3 above)
Comment (yellowjaguar5) [reply source/docs/hardware/sensors/encoders-hardware.rst]: Added in 8330140.

**Pull request merged**